src/components/App/index.js

```diff
@@ -2,50 +2,37 @@ import { h, Component } from 'preact'
 import classNames from 'classnames'
 import { bindActionCreators } from 'redux'
 import { connect } from 'react-redux'
-import {
-  unboundActions,
-  store,
-  events,
-  selectors
-} from 'onfido-sdk-core'
-import {stepsToComponents} from './StepComponentMap'
+import { unboundActions, events } from 'onfido-sdk-core'
+import { steps, components } from './StepComponentMap'
 import Error from '../Error'
 
 const App = ({ websocketErrorEncountered, step, options, socket, ...otherProps }) => {
-    const stepDefaultOptions = {
-      socket,
-      step,
-      ...otherProps
-    }
-    const stepsToComponentsWithDefaultOptions = steps => stepsToComponents(stepDefaultOptions, steps)
-    const defaultSteps = ['welcome','document','face','complete']
-    const stepComponents = stepsToComponentsWithDefaultOptions(options.steps || defaultSteps)
+  const componentOptions = {
+    socket,
+    step,
+    ...otherProps
+  }
+  const defaultStepOptions = ['welcome','document','face','complete']
+  const stepOptions = options.steps || defaultStepOptions
+  const stepList = steps(stepOptions, componentOptions.documentType)
+
+  if (step + 1 >=  stepList.length) componentOptions.nextStep = finalStep
```

Comment:
Whenever nextStep would have been called. Assuming a capture step (document or live photo) is the last step it's called in the `Confirm` component's `confirmAction` callback